galsim/interpolatedimage.py

```diff
@@ -550,6 +546,182 @@ def __setstate__(self, d):
+
+        # Check edges of input Images to see if gsparams.maxk_threshold criterion is met.  We
+        # can't actually know the values outside of the provide Images, of course, so this is
+        # only a warning.
+        if gsparams is None:
+            maxk_threshold = galsim.GSParams().maxk_threshold
+        else:
+            maxk_threshold = gsparams.maxk_threshold
+        edge = np.zeros_like(real_kimage.array, dtype=bool)
+        edge[[0,-1],:] = True # top and bottom
+        edge[:,[0,-1]] = True # left and right
+        edge_max = np.max(np.abs(real_kimage.array[edge] + 1j*imag_kimage.array[edge]))
+        if (edge_max / real_kimage.array.max()) > maxk_threshold:
+            import warnings
+            warnings.warn(
+                "Provided Real and Imag kimages do not meet GSParams.maxk_threshold criterion.")
+
+        if stepk is None:
+            stepk = real_kimage.scale
+        else:
+            if stepk < real_kimage.scale:
+                import warnings
+                warnings.warn(
+                    "Provided stepk is smaller than kimage.scale; overriding with kimage.scale.")
+                stepk = real_kimage.scale
+
+        self._real_kimage = real_kimage
+        self._imag_kimage = imag_kimage
+        self._stepk = stepk
+        self._gsparams = gsparams
+
+        # set up k_interpolant if none was provided by user, or check that the user-provided one
+        # is of a valid type
+        if k_interpolant is None:
+            self.k_interpolant = galsim.Quintic(tol=1e-4)
+        else:
+            self.k_interpolant = galsim.utilities.convert_interpolant(k_interpolant)
+
+        GSObject.__init__(self, galsim._galsim.SBInterpolatedKImage(
+            self._real_kimage.image, self._imag_kimage.image,
+            self._real_kimage.scale, self._stepk, self.k_interpolant, gsparams))
+
+    def __repr__(self):
+        return ('galsim.InterpolatedKImage(%r, %r, %r, stepk=%r, gsparams=%r)')%(
+            self._real_kimage, self._imag_kimage, self.k_interpolant,
+            self._stepk, self._gsparams)
+
+    def __str__(self): return 'galsim.InterpolatedKImage(real_kimage=%s)'%(
+            self._real_kimage)
+
+    def __getstate__(self):
+        # The SBInterpolatedKImage and the SBProfile both are picklable, but they are pretty
+        # inefficient, due to the large images being written as strings.  Better to pickle
+        # the intermediate products and then call init again on the other side.  There's still
+        # an image to be pickled, but at least it will be through the normal pickling rules,
+        # rather than the repr.
+        d = self.__dict__.copy()
+        del d['SBProfile']
+        return d
+
+    def __setstate__(self, d):
+        self.__dict__ = d
+        self.__init__(self._real_kimage, self._imag_kimage, k_interpolant=self.k_interpolant,
+                      stepk=self._stepk, gsparams=self._gsparams)
```

Comment:
I don't think you want to repeat all the tests that are in the `__init__` function.  Just need to rebuild the SBProfile:
```
self.__dict__ = d
self.SBProfile = galsim._galsim.SBInterpolatedKImage(
            self._real_kimage.image, self._imag_kimage.image,
            self._real_kimage.scale, self._stepk, self.k_interpolant, self._gsparams))
```